cmd/ical-relay/api.go

```diff
@@ -44,6 +44,97 @@ func calendarlistApiHandler(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, string(caljson)+"\n")
 }
 
+// Path: /api/profiles
+func profileApiHandler(w http.ResponseWriter, r *http.Request) {
+	requestLogger := log.WithFields(log.Fields{"client": GetIP(r), "api": r.Method + " " + r.URL.Path})
+	requestLogger.Infoln("New API-Request!")
+
+	token := r.Header.Get("Authorization")
+	if !checkSuperAuthorization(token) {
+		requestLogger.Warnln("Authorization not successful!")
+		w.WriteHeader(http.StatusUnauthorized)
+		fmt.Fprint(w, "Unauthorized!\n")
+		return
+	}
+
+	type profileJson struct {
+		Name          string   `json:"name"`
+		Sources       []string `json:"sources"`
+		Public        bool     `json:"public"`
+		ImmutablePast bool     `json:"immutable_past"`
+	}
+
+	switch r.Method {
+	case http.MethodPost:
+		// Create new profile
+		var newProfile profileJson
+
+		decoder := json.NewDecoder(r.Body)
+		err := decoder.Decode(&newProfile)
+		if err != nil {
+			requestLogger.Errorln(err)
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprint(w, "Error decoding json: "+err.Error()+"\n")
+			return
+		}
+
+		if conf.profileExists(newProfile.Name) {
+			requestLogger.Errorln("Profile already exists!")
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprint(w, "Error: Profile already exists!\n")
+			return
+		}
+
+		conf.addProfile(newProfile.Name, newProfile.Sources, newProfile.Public, newProfile.ImmutablePast)
+
+		requestLogger.Infoln("Created new profile: " + newProfile.Name)
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, "Created new profile: "+newProfile.Name+"\n")
+
+	case http.MethodPatch:
+		// Update profile
+		var newProfile profileJson
+
+		decoder := json.NewDecoder(r.Body)
+		err := decoder.Decode(&newProfile)
+		if err != nil {
+			requestLogger.Errorln(err)
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprint(w, "Error decoding json: "+err.Error()+"\n")
+			return
+		}
+
+		if !conf.profileExists(newProfile.Name) {
+			requestLogger.Errorln("Profile doenst exist!")
+			w.WriteHeader(http.StatusBadRequest)
+			fmt.Fprint(w, "Error: Profile doesnt exist!\n")
+			return
+		}
+
+		conf.editProfile(newProfile.Name, newProfile.Sources, newProfile.Public, newProfile.ImmutablePast)
+
+		requestLogger.Infoln("Edited new profile: " + newProfile.Name)
+		w.WriteHeader(http.StatusOK)
+		fmt.Fprint(w, "Edited new profile: "+newProfile.Name+"\n")
+
+	case http.MethodDelete:
+		// Delete profile
+		profileName := r.URL.Query().Get("name")
```

Comment:
I would like this to be solved without a query parameter, but this would require doing
PUT /api/profiles/profileName
DELETE /api/profiles/profileName
(optional POST /api/profiles)
(optional PATCH /api/profiles/profileName)